controller/linodemachine_controller.go

```diff
@@ -378,6 +388,67 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 	return linodeInstance, nil
 }
 
+func (r *LinodeMachineReconciler) configureDisks(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	linodeInstanceID int,
+) error {
+	// get the default instance config
+	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, linodeInstanceID, &linodego.ListOptions{})
+	if err != nil || len(configs) == 0 {
+		logger.Error(err, "Failed to list instance configs")
+
+		return err
+	}
+	instanceConfig := configs[0]
+
+	// carve out half the root disk space to be used for the etcd disk
+	rootDiskID := instanceConfig.Devices.SDA.DiskID
+	rootDisk, err := machineScope.LinodeClient.GetInstanceDisk(ctx, linodeInstanceID, rootDiskID)
+	if err != nil {
+		logger.Error(err, "Failed to get root disk for instance")
+
+		return err
+	}
+	//nolint:gomnd // cutting disk in half
+	diskSize := rootDisk.Size / 2
```

Comment:
Should this be bounded? At least to the recommended limit (8GB)? https://etcd.io/docs/v3.3/dev-guide/limit/#storage-size-limit

Reply:
could we perhaps set the rootdisk to disksize-8GB and the etcd-data to 8gb?

Reply:
Ah, it looks like we're already setting the quota in ETCD to 8GB here: https://github.com/linode/cluster-api-provider-linode/blob/b33f0355d1666bc4ba5b29ab9c32455988ddb8b9/templates/flavors/clusterclass-kubeadm/kubeadm-controlplane-template.yaml#L37-L38

Reply:
I've configured this to be 8 GB and updated the etcd doc to note this